File: 1456.py
import math

def prime_list(n):
    sieve = [True] * n

    m = int(n ** 0.5)+1
    for i in range(2, m + 1):
        if sieve[i] == True:           
            for j in range(i+i, n, i): 
                sieve[j] = False

    return [i for i in range(2, n) if sieve[i] == True]

a,b = map(int, input().split())
ans =0
plist = prime_list(b)

for k in plist:
    tmp = k*k

    while tmp <= b:
        if a<= tmp <= b:
            ans +=1
        tmp = tmp*k

print(ans)

File: 1456.cpp
#include <iostream>
#include <stdlib.h>
#include <math.h>

using namespace std;

bool isPrime(long long n){
    if(n==2 || n==3)
        return true;
    else if (n%2==0 || n==1)
        return false;
    
    for (uint64_t i=3; i <  uint64_t(sqrtl(n)+1); i+=2){
        if(n%i == 0)
            return false;
    }
    return true;
}

long long nextPrime(long long n){
    if(n==2)
        return 3;
    long long tmp = n+1;
    while(!isPrime(tmp))
        tmp+=2;
    return tmp;
}

int main(){
    long ans = 0, n;
    long long a,b;
    cin >> a >> b;
    n = nextPrime(1);
    long tar = long(sqrt(b)+1);
    
    while(n<tar){
        long long tmp = n*n;
        while(tmp<=b){
            if(tmp >= a)
                ans += 1;
            tmp = tmp * n;
        }
        n = nextPrime(n);
    }
    cout << ans << endl;

    return 0;
}